### PR TITLE
Fixed InputManager.asset file growing in size on each Reset call

### PR DESCRIPTION
**Pull request by lyndon-unity**
### Description

Reset now clears out old Input Action Assets and Input Action References first to avoid the InputManager.asset file growing in size over each reset

For bug https://jira.unity3d.com/browse/ISX-1839 

### Changes made

Spinning through contents of InputManager.asset to remove Input Action Assets and Input Action References first 
Then using GetOrCreate to create a new version (to make sure only 1 added, if GetOrCreate called in a callback elsewhere - which it is due to AssetDatabase.SaveAssets call)

### Checklist

Before review:

- [X] Changelog entry added.
- [ ] Tests added/changed, if applicable.
- None added on this instance, but opportunity to add one for this

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

### CODE REVIEW

File: Packages/com.unity.inputsystem/CHANGELOG.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ however, it has to be formatted properly to pass verification tests.
 - Fixed issue where composite part dropdown manipulates binding path and leaves composite part field unchanged.
 - Fixed lingering highlight effect on Save Asset button after clicking.
 - Fixed missing name in window title for Input Action assets.
+- Fixed InputManager.asset file growing in size on each Reset call (ISX-1839)
```
Comment (ekcoh) [inline Packages/com.unity.inputsystem/CHANGELOG.md]: Minor: Missing full-stop in end. Referencing internal case ID for a non-publicly visible issue.

**Review by ekcoh — APPROVED** [review]
LGTM, a minor comment about changelog.

**Comment by ekcoh** [comment]
Seems like PR got hit by CI instability, rerun tests?

File: Packages/com.unity.inputsystem/CHANGELOG.md (hunk 1 above)
Comment (lyndon-unity) [reply Packages/com.unity.inputsystem/CHANGELOG.md]: Changed (online in githib editor ;) )
Note: Some prior change log entries also seem to have some internal bug references but I've not changed those in this PR

**Comment by lyndon-unity** [comment]
> Seems like PR got hit by CI instability, rerun tests?

Re-running with minor changelog update.

File: Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs [hunk 2]
```diff
@@ -147,7 +149,46 @@ internal static void CheckForDefaultUIActionMapChanges()
...
+            DeleteActionAsset();
+
+            // Note the window can refresh between deletion and creation so we should only re-create if not present
+            GetOrCreate();
+        }
+
+        /// <summary>
+        /// Delete project wide input actions
+        /// </summary>
+        /// <param name="asset"></param>
```
Comment (ritamerkl) [inline Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs]: parameter left over in comment

File: Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs [hunk 3]
```diff
@@ -147,7 +149,46 @@ internal static void CheckForDefaultUIActionMapChanges()
             }
         }
 
-        #endif
+#endif
+        /// <summary>
+        /// Reset project wide input actions asset
+        /// </summary>
+        /// <param name="asset"></param>
```
Comment (ritamerkl) [inline Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs]: unused param asset

**Review by ritamerkl — APPROVED** [review]
Just two little remarks on the comments, looks good, nice to have this fixed!

**Comment by lyndon-unity** [comment]
Params documentation fixed - and moved delete into the create so that when new data created any old data is always cleaned out. This catches an extra case.

File: Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs [hunk 4]
```diff
@@ -147,13 +150,44 @@ internal static void CheckForDefaultUIActionMapChanges()
...
+                    if (obj is InputActionReference)
+                    {
+                        var actionReference = obj as InputActionReference;
+                        actionReference.Set(null);
+                        AssetDatabase.RemoveObjectFromAsset(obj);
+                    }
+                    else if (obj is InputActionAsset)
+                    {
+                        AssetDatabase.RemoveObjectFromAsset(obj);
+                    }
```
Comment (ekcoh) [inline Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs:182]: What happens if there is something here that isn't an InputActionReference or an InpuitAction asset?
I recommend we either throw or log an error or alternatively change the last else if into an unconditional else which would remove anything? Or is this designed to leave e.g. InputManager or any other root objects untouched?

**Review by ekcoh — APPROVED** [review]
Approving but have one outstanding questions around the intended behavior in the `DeleteActionAssetAndActionReferences()` function regarding intended behavior to consider before merge.

File: Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs (hunk 4 above)
Comment (lyndon-unity) [reply Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs:182]: Its designed to leave InputManager and others untouched

**Pull request merged**